src/db.rs

```diff
@@ -2866,6 +2881,127 @@ mod tests {
         .await
     }
 
+    #[tokio::test]
+    async fn test_recover_clock_tick_from_wal() {
+        let clock = Arc::new(TestClock::new());
+        let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+        let path = Path::from("/tmp/test_kv_store");
+
+        let db = Db::open_with_opts(
+            path.clone(),
+            test_db_options_with_clock(0, 1024, None, clock.clone()),
+            Arc::clone(&object_store),
+        )
+        .await
+        .unwrap();
+
+        clock.ticker.store(10, Ordering::SeqCst);
+        db.put(&[b'a'; 4], &[b'j'; 8])
+            .await
+            .expect("write batch failed");
+        clock.ticker.store(11, Ordering::SeqCst);
+        db.put(&[b'b'; 4], &[b'k'; 8])
+            .await
+            .expect("write batch failed");
+
+        // close the db to flush the manifest
```

Comment:
nit: comment seems off with an explicit flush below

Reply:
That was from before your fix 😉 now we don't need two